### DISCUSSION TITLE
cargo build error

### DISCUSSION THREAD

**Discussion by ExtSolTech** [fltk-rs/fltk-rs#1116] [Q&A]
Hi there,

I follow the building instructions to run examples, but i have a little problem.

git clone working
cargo build give me an error message:
![image](https://user-images.githubusercontent.com/44647576/150230658-028efc49-13c2-4158-bf40-10cf5b6f727b.png)

i have installed and wokring cargo, git and cmake:
![image](https://user-images.githubusercontent.com/44647576/150230885-424f566e-2bd4-4744-8304-d589b6f7295b.png)

Where can i write the fltk-bundled flag?
![image](https://user-images.githubusercontent.com/44647576/150231720-ec011f9e-3f47-4425-b48c-71a6466cd31b.png)

Sorry for my beginner questions, but this is my the second day in the Rust world

**Comment by MoAlyousef** [comment]
You would use the fltk-bundled feature when you create a new project using `cargo new app`, then `cd app` and modifiy the Cargo.toml file to:
```toml
[dependencies]
fltk = { version = "1.2", features = ["fltk-bundled"] }
```
But note that there are no bundles for x86 (i.e. 32 bit windows).

The error that's happening might be from toolchain mismatch. What's the output of `rustup show`?

**Reply by ExtSolTech** [reply]
I dont't want to create now a new app, i only want to build and run examples:
[https://github.com/fltk-rs/fltk-rs/tree/master/fltk/examples](url)

here is my output: 
![image](https://user-images.githubusercontent.com/44647576/150294726-bc4ed4ba-1db5-4c44-875a-5038828c8f8d.png)

**Reply by MoAlyousef** [reply]
Ok, so what’s the output of `cargo clean && cargo build -vv —example hello`?

**Reply by ExtSolTech** [reply]
![image](https://user-images.githubusercontent.com/44647576/150298028-2c79f296-7a04-455b-a914-6ff3fe8d0830.png)

**Reply by ExtSolTech** [reply]
C:\Users\ExtSol\Downloads\fltk-rs>cargo clean && cargo build --verbose --example hello
   Compiling cc v1.0.72
   Compiling crossbeam-utils v0.8.6
   Compiling cfg-if v1.0.0
   Compiling lazy_static v1.4.0
   Compiling bitflags v1.3.2
   Compiling paste v1.0.6
   Compiling ttf-parser v0.14.0
     Running `rustc --crate-name cc --edition=2018 C:\Users\ExtSol\.cargo\registry\src\github.com-1ecc6299db9ec823\cc-1.0.72\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 -C metadata=2f018b83f80fbfcb -C extra-filename=-2f018b83f80fbfcb --out-dir C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps -L dependency=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps --cap-lints allow`
     Running `rustc --crate-name build_script_build --edition=2018 C:\Users\ExtSol\.cargo\registry\src\github.com-1ecc6299db9ec823\crossbeam-utils-0.8.6\build.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --cfg "feature=\"lazy_static\"" --cfg "feature=\"std\"" -C metadata=3cbe8f6717622a46 -C extra-filename=-3cbe8f6717622a46 --out-dir C:\Users\ExtSol\Downloads\fltk-rs\target\debug\build\crossbeam-utils-3cbe8f6717622a46 -L dependency=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps --cap-lints allow`
     Running `rustc --crate-name cfg_if --edition=2018 C:\Users\ExtSol\.cargo\registry\src\github.com-1ecc6299db9ec823\cfg-if-1.0.0\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 -C metadata=29c580bedc5fe890 -C extra-filename=-29c580bedc5fe890 --out-dir C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps -L dependency=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps --cap-lints allow`
     Running `rustc --crate-name lazy_static C:\Users\ExtSol\.cargo\registry\src\github.com-1ecc6299db9ec823\lazy_static-1.4.0\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 -C metadata=cbedf92a9033d70b -C extra-filename=-cbedf92a9033d70b --out-dir C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps -L dependency=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps --cap-lints allow`
     Running `rustc --crate-name bitflags --edition=2018 C:\Users\ExtSol\.cargo\registry\src\github.com-1ecc6299db9ec823\bitflags-1.3.2\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg "feature=\"default\"" -C metadata=32a338a3f8049abe -C extra-filename=-32a338a3f8049abe --out-dir C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps -L dependency=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps --cap-lints allow`
     Running `rustc --crate-name paste --edition=2018 C:\Users\ExtSol\.cargo\registry\src\github.com-1ecc6299db9ec823\paste-1.0.6\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type proc-macro --emit=dep-info,link -C prefer-dynamic -C embed-bitcode=no -C debuginfo=2 -C metadata=99ac3daba3de08fa -C extra-filename=-99ac3daba3de08fa --out-dir C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps -L dependency=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps --extern proc_macro --cap-lints allow`
     Running `rustc --crate-name ttf_parser --edition=2018 C:\Users\ExtSol\.cargo\registry\src\github.com-1ecc6299db9ec823\ttf-parser-0.14.0\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg "feature=\"default\"" --cfg "feature=\"glyph-names\"" --cfg "feature=\"opentype-layout\"" --cfg "feature=\"std\"" --cfg "feature=\"variable-fonts\"" -C metadata=659b4c63c8c983c4 -C extra-filename=-659b4c63c8c983c4 --out-dir C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps -L dependency=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps --cap-lints allow`
     Running `C:\Users\ExtSol\Downloads\fltk-rs\target\debug\build\crossbeam-utils-3cbe8f6717622a46\build-script-build`
   Compiling cmake v0.1.48
     Running `rustc --crate-name cmake C:\Users\ExtSol\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.48\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 -C metadata=c8472dba61ac8ed2 -C extra-filename=-c8472dba61ac8ed2 --out-dir C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps -L dependency=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps --extern cc=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps\libcc-2f018b83f80fbfcb.rmeta --cap-lints allow`
     Running `rustc --crate-name crossbeam_utils --edition=2018 C:\Users\ExtSol\.cargo\registry\src\github.com-1ecc6299db9ec823\crossbeam-utils-0.8.6\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg "feature=\"lazy_static\"" --cfg "feature=\"std\"" -C metadata=590ba6594be30e82 -C extra-filename=-590ba6594be30e82 --out-dir C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps -L dependency=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps --extern cfg_if=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps\libcfg_if-29c580bedc5fe890.rmeta --extern lazy_static=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps\liblazy_static-cbedf92a9033d70b.rmeta --cap-lints allow`
   Compiling crossbeam-channel v0.5.2
     Running `rustc --crate-name crossbeam_channel --edition=2018 C:\Users\ExtSol\.cargo\registry\src\github.com-1ecc6299db9ec823\crossbeam-channel-0.5.2\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg "feature=\"crossbeam-utils\"" --cfg "feature=\"default\"" --cfg "feature=\"std\"" -C metadata=d217e9432629521a -C extra-filename=-d217e9432629521a --out-dir C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps -L dependency=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps --extern cfg_if=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps\libcfg_if-29c580bedc5fe890.rmeta --extern crossbeam_utils=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps\libcrossbeam_utils-590ba6594be30e82.rmeta --cap-lints allow`
   Compiling fltk-sys v1.2.28 (C:\Users\ExtSol\Downloads\fltk-rs\fltk-sys)
     Running `rustc --crate-name build_script_main --edition=2018 fltk-sys\build/main.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --cfg "feature=\"default\"" -C metadata=693e8d60ed7483e8 -C extra-filename=-693e8d60ed7483e8 --out-dir C:\Users\ExtSol\Downloads\fltk-rs\target\debug\build\fltk-sys-693e8d60ed7483e8 -C incremental=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\incremental -L dependency=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps --extern cmake=C:\Users\ExtSol\Downloads\fltk-rs\target\debug\deps\libcmake-c8472dba61ac8ed2.rlib`
     Running `C:\Users\ExtSol\Downloads\fltk-rs\target\debug\build\fltk-sys-693e8d60ed7483e8\build-script-main`
error: failed to run custom build command for `fltk-sys v1.2.28 (C:\Users\ExtSol\Downloads\fltk-rs\fltk-sys)`

Caused by:
  process didn't exit successfully: `C:\Users\ExtSol\Downloads\fltk-rs\target\debug\build\fltk-sys-693e8d60ed7483e8\build-script-main` (exit code: 101)
  --- stdout
  cargo:rerun-if-changed=build/android.rs
  cargo:rerun-if-changed=build/bundled.rs
  cargo:rerun-if-changed=build/link.rs
  cargo:rerun-if-changed=build/main.rs
  cargo:rerun-if-changed=build/source.rs
  cargo:rerun-if-changed=build/utils.rs
  cargo:rerun-if-env-changed=CC
  cargo:rerun-if-env-changed=CXX
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-changed=cfltk/CMakeLists.txt
  cargo:rerun-if-changed=cfltk/include/cfl.h
  cargo:rerun-if-changed=cfltk/include/cfl_widget.h
  cargo:rerun-if-changed=cfltk/include/cfl_group.h
  cargo:rerun-if-changed=cfltk/include/cfl_input.h
  cargo:rerun-if-changed=cfltk/include/cfl_window.h
  cargo:rerun-if-changed=cfltk/include/cfl_button.h
  cargo:rerun-if-changed=cfltk/include/cfl_box.h
  cargo:rerun-if-changed=cfltk/include/cfl_menu.h
  cargo:rerun-if-changed=cfltk/include/cfl_dialog.h
  cargo:rerun-if-changed=cfltk/include/cfl_valuator.h
  cargo:rerun-if-changed=cfltk/include/cfl_browser.h
  cargo:rerun-if-changed=cfltk/include/cfl_misc.h
  cargo:rerun-if-changed=cfltk/include/cfl_text.h
  cargo:rerun-if-changed=cfltk/include/cfl_image.h
  cargo:rerun-if-changed=cfltk/include/cfl_draw.h
  cargo:rerun-if-changed=cfltk/include/cfl_table.h
  cargo:rerun-if-changed=cfltk/include/cfl_surface.h
  cargo:rerun-if-changed=cfltk/include/cfl_printer.h
  cargo:rerun-if-changed=cfltk/include/cfl_utils.h
  cargo:rerun-if-changed=cfltk/include/cfl_macros.h
  cargo:rerun-if-changed=cfltk/include/cfl_lock.h
  cargo:rerun-if-changed=cfltk/include/cfl_widget.hpp
  cargo:rerun-if-changed=cfltk/src/cfl_lock.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_new.cpp
  cargo:rerun-if-changed=cfltk/src/cfl.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_widget.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_group.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_window.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_button.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_box.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_menu.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_dialog.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_valuator.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_browser.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_misc.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_text.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_image.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_input.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_draw.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_table.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_tree.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_surface.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_printer.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_font.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_utils.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_nswindow.m
  cargo:rerun-if-changed=cfltk/FL_FLex/FL_Flex.H
  cargo:rerun-if-changed=cfltk/FL_FLex/FL_Flex.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_vec.hpp
  CMAKE_TOOLCHAIN_FILE_x86_64-pc-windows-msvc = None
  CMAKE_TOOLCHAIN_FILE_x86_64_pc_windows_msvc = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_x86_64-pc-windows-msvc = None
  CMAKE_GENERATOR_x86_64_pc_windows_msvc = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_x86_64-pc-windows-msvc = None
  CMAKE_PREFIX_PATH_x86_64_pc_windows_msvc = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_x86_64-pc-windows-msvc = None
  CMAKE_x86_64_pc_windows_msvc = None
  HOST_CMAKE = None
  CMAKE = None

  --- stderr
  error: patch failed: src/drivers/Android/Fl_Android_System_Driver.H:25
  error: src/drivers/Android/Fl_Android_System_Driver.H: patch does not apply
  thread 'main' panicked at '

  couldn't determine visual studio generator
  if VisualStudio is installed, however, consider running the appropriate vcvars script before building this crate
  ', C:\Users\ExtSol\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.48\src\lib.rs:878:25
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

**Reply by MoAlyousef** [reply]
To build from source on windows-msvc, you need msvc and the windows sdk, setup described here:
https://fltk-rs.github.io/fltk-book/Setup.html#detailed-setup

And you can’t use the fltk-bundled feature with the repo. Only on external projects.

**Reply by ExtSolTech** [reply]
i have installed msvc toolchain
![image](https://user-images.githubusercontent.com/44647576/150299951-a53384e2-3d4b-4deb-9965-df9b9b13243a.png)

**Reply by MoAlyousef** [reply]
It might be cmake isn’t able to find the generator. Try building from the native tools command prompt. In the start menu, start typing “native” and it should show. Change into the fltk-rs directory and run cargo clean then cargo build —example hello

**Reply by ExtSolTech** [reply]
yeaaahhh thank you for your help and time
![image](https://user-images.githubusercontent.com/44647576/150306578-c852acb9-8376-4a0c-8852-ddc706dc8e8f.png)

**Reply by MoAlyousef** [reply]
No problem 😉
